src/filesystem/impls/filer/lib/ImageResizer.js

```diff
@@ -0,0 +1,88 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define */
+define(function (require, exports, module) {
+    "use strict";
+
+    var pica = require("Pica")();
+
+    var Content = require("filesystem/impls/filer/lib/content");
+    var Path = require("filesystem/impls/filer/FilerUtils").Path;
+    var decodePath = require("filesystem/impls/filer/FilerUtils").decodePath;
+    var base64ToBuffer = require("filesystem/impls/filer/FilerUtils").base64ToBuffer
+
+    function ImageResizer(path, data) {
+        this.path = decodePath(path);
+        this.type = Content.mimeFromExt(Path.extname(this.path));
+        this.blob = new Blob([data], {type: this.type});
+        this.url = URL.createObjectURL(this.blob);
+        this.canvas = document.createElement('canvas');
+    }
+    ImageResizer.prototype.cleanup = function() {
+        URL.revokeObjectURL(this.url);
+    };
+    ImageResizer.prototype.resize = function(callback) {
+        var self = this;
+        var scalingFactor = 0.75;
+        var buffer;
+        var passes = 0;
+
+        function finish(err, buffer) {
+            self.cleanup();
+            callback(err, buffer);
+        }
+
+        function resizePass() {
+            console.log("resizePass", passes, scalingFactor);
+
+            if(passes++ > 5) {
+                finish(null, buffer);
+            }
+
+            self.canvas.width = img.width * scalingFactor;
+            self.canvas.height = img.height * scalingFactor;
+
+            console.log("width", self.canvas.width, "height", self.canvas.height);
+
+            pica.resize(img, self.canvas)
```

Comment:
honestly, if we already have a `self.canvas` then we don't need pica for this, since this functionality is explicitly [baked into the Canvas2D API](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/drawImage). We just need two lines rather than one:
```
var context = canvas.getContext('2d");
context.drawImage(img, 0, 0, self.canvas.width, self.canvas.height);
```
and that's the resizing achieved without an additional library.